src/nimblepkg/download.nim

```diff
@@ -29,19 +22,6 @@ proc doCheckout(meth: DownloadMethod, downloadDir, branch: string) =
     cd downloadDir:
       doCmd("hg checkout " & branch)
 
-proc doPull(meth: DownloadMethod, downloadDir: string) =
-  case meth
-  of DownloadMethod.git:
-    doCheckout(meth, downloadDir, "")
-    cd downloadDir:
-      doCmd("git pull")
-      if existsFile(".gitmodules"):
-        doCmd("git submodule update")
-  of DownloadMethod.hg:
-    doCheckout(meth, downloadDir, "default")
-    cd downloadDir:
-      doCmd("hg pull")
-
```

Comment:
While these are not used currently I think they're worth keeping, they might be useful soon.

Reply:
Do you have something specific in mind? I will return them if so, but my issue with them was that just recently, when I had added `local` download method, I was required to maintain their code also, despite that they are not being used. I'm expecting soon to work on adding `tarball` download method and such a maintenance will be required again.

Reply:
yes, the plan is to make Nimble manage git repos instead of re-cloning packages at different commit hashes all the time.

Reply:
Reverted in 057505b.